src/server/standalone.js

```diff
@@ -194,9 +195,19 @@ function StandAloneServer(gmeConfig) {
             }
         });
 
-        __httpServer.listen(gmeConfig.server.port, function () {
+        // something link
+        // https://github.com/rubenv/node-autoquit
+        // should be considered to shut the service down if it
+        // is not getting much use.
+
+        // https://github.com/herzi/systemd-socket
+        // https://nodejs.org/api/http.html#http_server_listen_handle_callback
+        __httpServer.listen(systemdSocket(0) || gmeConfig.server.port, function () {
```

Comment:
Sorry I took so long to get back to this.
The problem with the approach you advocate is that systemSocket(0) does not return a number.
systemSocket(0) returns an object of the form '{fd: 3}'.
This should be simple enough to fix.
The problem is in the validator, it does not like an argument of the form '{fd: 3}'.
I hacked webgme/config/validator.js:194
// assertNumber('config.server.port', config.server.port);
This allows the file descriptor adoption.